packages/lifi-adapter/src/service.ts

```
@@ -0,0 +1,367 @@
+            destination: route.destination,
+            amountIn: quote.estimate.fromAmount,
+            amountOut: quote.estimate.toAmount,
+            effectiveRate,
+            totalFeesUsd,
+            quotedAt: new Date().toISOString(),
+          });
+        } catch (error) {
+          console.error('Failed to get rate for route:', { error: error instanceof Error ? error.message : 'Unknown error' });
+          rates.push(this.createFallbackRate(route, notional));
+        }
+      }
+    }
+
+    return rates;
+  }
+
+
+
+  /**
+   * Probe liquidity depth using binary search for accurate thresholds
+   */
+  private async getLiquidityDepth(routes: Array<{ source: AssetType; destination: AssetType }>): Promise<LiquidityDepthType[]> {
+    if (!routes?.length) {
+      throw new Error('Routes are required for liquidity depth calculation');
+    }
+
+    const liquidity: LiquidityDepthType[] = [];
+
+    for (const route of routes) {
+      if (!route?.source || !route?.destination) {
+        console.warn('Invalid route structure for liquidity probing, skipping');
+        continue;
+      }
+
+      try {
+        const thresholds = [];
+        
+        // Binary search for 50bps and 100bps thresholds
+        for (const slippageBps of [50, 100]) {
+          try {
+            const maxAmountIn = await LiquidityProber.findMaxLiquidity(
+              this.baseUrl,
+              route,
+              slippageBps,
+              3 // Limited iterations for performance
+            );
+            
+            thresholds.push({
+              maxAmountIn,
+              slippageBps,
+            });
+          } catch (error) {
+            console.error('Liquidity probing failed:', { slippageBps, error: error instanceof Error ? error.message : 'Unknown error' });
+            // Fallback to conservative estimate
+            thresholds.push({
+              maxAmountIn: '1000000', // 1 USDC
+              slippageBps,
+            });
+          }

```

No fallback. Remove this, throw an error. 

It looks like this may be failing in the snapshot, we will have to investigate.

Can you share links to relevant docs for determining available liquidity?

---

Liquidity is determined indirectly by probing the /quote endpoint ([https://docs.li.fi/api-reference/get-a-quote-for-a-token-transfer-1](vscode-file://vscode-app/c:/Users/UseR/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html)).
How it works:
•	The slippage parameter allows us to set tolerance thresholds (50bps/100bps)
•	We test increasing fromAmount values until /quote returns success/failure
•	Binary search finds the maximum feasible input at each slippage threshold
•	Success = liquidity exists at that amount; Failure = no liquidity

I've removed the fallback - now when probing fails, we throw an error instead of returning stale data.
(https://github.com/Psianturi/lifi-adapter-for-near/commit/2ef72b925e3b9ab37e252acb727de3ad9fc36edc)